histexpand: add `!N` and `!?string?` commands (#6051)
tldr-pages/tldr: pages/common/histexpand.md

> Reuse and expand the shell history in `sh`, `bash`, `zsh`, `rbash` and `ksh`.
> More information: <https://www.gnu.org/software/bash/manual/html_node/History-Interaction>.

- Run the previous command:

> `!!`

Reviewer: Despite the fact that this example is simple, I think it's one of the most important on the page, as it introduces the user to the concept of history expansion with something they may already be familiar with:

![](https://imgs.xkcd.com/comics/sandwich.png)

Reviewer 2: It's sometimes hard to keep in mind for whom tldr is indented, so I have to agree. I think it would be best if we would make a poll, if and which other command to remove.

(BTW, how do some people have the ability to always find the right xkcd in the right moment?)

Reviewer: \* indented → intended

I'd prefer no polls here - generally speaking we try to talk it out instead, so everyone's happy with the outcome. @ waldyrious explained this in a previous issue/PR a while back

Reviewer 2: Makes sense. So, how if we romevo one of those:
* `!-n` - I know I wanted to keep it at the beginning but I'm really sure that it's really useful anymore
* One of either `!^` or `!$`, since I think they're both rarely used. Maybe we could even combine them somehow.

Reviewer: What about removing `!^`? `!$` can be useful  e.g. in this situation:

```
nano path/to/file
node !$
```

Reviewer 2: Any other opinions? @bl-ue, @navarroaxel, @CleanMachine1

Reviewer 3: I think we can merge `!!` and `sudo !!` but otherwise remove `!^` is a good option.

Reviewer 3: I think we can merge `!!` and `sudo !!` but otherwise remove `!^` is a good option.

Reviewer 2: That's also an option. If we merged the first to into
```
- Run the previous command as root (`!!` replaces the previous command):

sudo !!
```
we'd solve the problem while still explaining the basic function of `!!` and not deleting any example.

Reviewer: Ah, that works @navarroaxel @marchersimon - let's do that.

Reviewer 3: @jopefd could you apply this change?

Reviewer 2: I made a [suggestion](https://github.com/tldr-pages/tldr/pull/6051/files#r657176121) for that

> - Run the previous command as root:
> 
> `sudo !!`

Reviewer 2: ```suggestion
- Run the previous command as root (`!!` replaces the previous command):

`sudo !!`
```

> Reuse and expand the shell history in `sh`, `bash`, `zsh`, `rbash` and `ksh`.
> More information: <https://www.gnu.org/software/bash/manual/html_node/History-Interaction>.

> - Run the previous command as root (`!!` replaces the previous command):

Reviewer: Perhaps this will read better?
```suggestion
- Run the previous command as root (`!!` is replaced by the previous command):
```

`{{command}} !^`

- Run the Nth command of the history:

> `!{{N}}`

Reviewer 2: ```suggestion
`!{{n}}`
```
We use `n` all over the page and I think it's more universal for that purpuse.

`{{command}} !^`

- Run the Nth command of the history:

`!{{N}}`

- Run the command `n` lines back in the history:

`!-{{n}}`

> - Run the most recent command starting with `string`:

Reviewer 4: This line should probably update too.  How about
```suggestion
- Run the most recent command containg with `string`:
```

Reviewer 3: `containing` @bl-ue ?

Author: yeah, @bl-ue is right

man history:
_!?string[?] Refer to the most recent command preceding the current position in the history list containing string.  The trailing ? may be omitted if string is followed immediately by a newline._

`{{command}} !^`

- Run the Nth command of the history:

`!{{N}}`

- Run the command `n` lines back in the history:

`!-{{n}}`

- Run the most recent command starting with `string`:

> `!{{string}}`
> `!?{{string}}?`

Reviewer 4: One of these has to be removed.  Let's remove the old one
```suggestion
`!?{{string}}?`
```
